src/libraries/Microsoft.Extensions.Configuration.Binder/gen/Specs/Types/TypeSpec.cs

```diff
@@ -12,15 +12,16 @@ internal abstract record TypeSpec
     {
         public TypeSpec(ITypeSymbol type)
         {
-            Namespace = type.ContainingNamespace?.ToDisplayString();
-            DisplayString = type.ToMinimalDisplayString();
-            Name = (Namespace is null ? string.Empty : Namespace + ".") + DisplayString.Replace(".", "+");
+            (Namespace, DisplayString, Name) = type.GetTypeName();
+            AssemblyQualifiedName = type.ContainingAssembly + type.ToDisplayString(SymbolDisplayFormat.FullyQualifiedFormat);
```

Comment:
The inclusion of the non-auto implemented properties (all the `abstract` ones) means this type is likely not fully value equitable as these properties will not be used in the equality calculation. I'd be suspicious of this use of `record`.